src/targets/gpu/include/migraphx/gpu/hip.hpp

```diff
@@ -144,17 +144,16 @@ struct hip_copy_from_gpu
         check_shapes{inputs, *this}.has(1, 2);
         return inputs.at(0);
     }
-    argument
-    compute(context& ctx, const shape& output_shape, const std::vector<argument>& args) const
+    argument compute(context& ctx, const shape&, const std::vector<argument>& args) const
     {
         if(args.size() == 1)
         {
-            argument result = allocate_gpu(output_shape, true);
+            argument result = allocate_gpu(args.front().get_shape(), true);
             gpu_copy(ctx, args[0], result);
             return result;
         }
         copy_from_gpu(ctx, args[0], args[1]);
-        return args[1];
+        return {args[0].get_shape(), args[1].data()};
```

Comment:
This should be `args[1]`. This can cause a dangling reference, since the result will not own the memory.

Reply:
This change is to support the dynamic output shape, which is required by the nonzero operator.
The scenario is: when we call the compute_shape method, we set the output shape to a max possible value, then in the compute method, if the actual output has less elements, i.e., a different shape, we will change the output shape to that from the compute method. 

Then in the gpu_copy, we need to do the same thing. In other words, if number of elems in the input is less than that of the output, we can still do the copy and change the output shape to be the same as the input. 

This change will not hurt any scenario that we can supported for now, and it will support more scenarios like the nonzero and the loop operators. But there are still scenarios, in whcih it will fail  in compute method of some operators.

Reply:
The compute method should return the same shape as the `compute_shape`.

Reply:
I would like to assert that this is true, but there are bugs in the `int8_pack_args` that need to be fixed first.